### DISCUSSION TITLE
Multiple data source, one datatable

### DISCUSSION THREAD

**Discussion by taufiqridha** [yajra/laravel-datatables#3033] [Q&A] [answered]
Hi is there anyone has successfully implement multiple data source to one datatable ? 
So let say we want 3 models / database / eloquent to display in one datatable...

any advice would be thankful
Taufiq

**Comment by yajra** [comment]
You can use DataTableHtml.

```bash
php artisan datatables:make Demo --builder
```

This will create `DemoDataTableHtml` that you can reuse on any page by calling `DemoDataTableHtml::make()`.

### Controller

```php
    public function index(UsersDataTable $dataTable)
    {
        $demo = DemoDataTableHtml::make();
        
        return $dataTable->render('users::index', compact('demo'));
    }
```

### View

```blade
{{ $dataTable->table() }}
{{ $demo->table() }}

@push('scripts')
{{ $dataTable->scripts() }}
{{ $demo->scripts() }}
@endpush
```

**Reply by yajra** [reply]
Oh just realized my mistake, the question is multiple sources and not multiple dataTable :)

**Comment by yajra** [comment, marked as answer]
For multiple data sources, we usually use `UNION` SQL.

**Reply by yajra** [reply]
If sources are from different databases, you would need to use one big collection instead. But note the performance issue for bulk data.
